src/serialize-spec.ts

```diff
@@ -188,6 +204,35 @@ function serializeNode(node: Node): any {
   }
 }
 
+function serializeInputContext(
+  context: ZkAppInputContext | HttpsInputContext
+): {
+  type: 'zk-app' | 'https';
+  presentationCircuitVKHash: ReturnType<typeof serializeProvable>;
+  action: ReturnType<typeof serializeProvable> | string;
+  serverNonce: ReturnType<typeof serializeProvable>;
+} {
+  if ('action' in context && typeof context.action === 'string') {
```

Comment:
nit: it would be cleaner to pass the `request.type` to this function, so that you can switch on that and don't have to inspect the action type here